stock_inventory_barcode/wizard/stock_inventory_barcode.py

```diff
@@ -0,0 +1,284 @@
+                        )
+                        % "\n".join([product.display_name for product in products]),
+                    }
+                }
+            else:
+                return {
+                    "warning": {
+                        "title": _("Error"),
+                        "message": _(
+                            "No stockable product found with this code as "
+                            "Barcode nor Internal Reference. You should select "
+                            "the right product manually."
+                        ),
+                    }
+                }
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        if self.product_id:
+            self.product_tracking = self.product_id.tracking
+            self.uom_id = self.product_id.uom_id
+            if not self.product_tracking or self.product_tracking == "none":
+                self.lot_id = False
+        else:
+            self.product_tracking = False
+            self.uom_id = False
+            self.lot_id = False
+            self.note = False
+
+    @api.onchange("product_id", "location_id", "lot_id", "uom_id")
+    def product_lot_loc_change(self):
+        res = {"warning": {}}
+        if self.product_id and self.location_id and self.uom_id:
+            if self.uom_id != self.product_id.uom_id:
+                self.uom_id = self.product_id.uom_id
+                res["warning"] = {
+                    "title": _("Error"),
+                    "message": _(
+                        "You cannot change the unit of measure of the "
+                        "product: it is been restored to '%s'."
+                    )
+                    % self.product_id.uom_id.name,
+                }
+            if not self.product_id.tracking or self.product_id.tracking == "none":
+                if self.lot_id:
+                    raise UserError(
+                        _(
+                            "Product '%s' is not tracked by lot/serial so the "
+                            "lot field must be empty. This should never "
+                            "happen."
+                        )
+                        % self.product_id.display_name
+                    )
+                self.update_wiz_screen(res)
+            elif self.lot_id:
+                self.update_wiz_screen(res)
+        return res
+
+    def update_wiz_screen(self, res):
+        silo = self.env["stock.inventory.line"]
+        ilines = silo.search(
+            [
+                ("inventory_id", "=", self.inventory_id.id),
+                ("product_id", "=", self.product_id.id),
+                ("location_id", "=", self.location_id.id),
+                ("prod_lot_id", "=", self.lot_id and self.lot_id.id or False),
+            ]
+        )
+        if len(ilines) == 1:
+            self.inventory_line_id = ilines[0]
+            self.change_qty = ilines[0].product_qty
+        elif len(ilines) > 1:
```

Comment:
following previous comment, it seems safe to remove code from this function from this line onwards (keeping the else block)